backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceBroadcastReceiver.java

```diff
@@ -0,0 +1,103 @@
+package backtraceio.library.breadcrumbs;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import backtraceio.library.enums.BacktraceBreadcrumbType;
+import backtraceio.library.logger.BacktraceLogger;
+
+public class BacktraceBroadcastReceiver extends BroadcastReceiver {
+
+    private BacktraceBreadcrumbs backtraceBreadcrumbs;
+
+    private static transient String LOG_TAG = BacktraceBroadcastReceiver.class.getSimpleName();
+
+    public BacktraceBroadcastReceiver(@NonNull BacktraceBreadcrumbs backtraceBreadcrumbs)
+    {
+        this.backtraceBreadcrumbs = backtraceBreadcrumbs;
+    }
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        String action = intent.getAction();
+        if (action == null)
+        {
+            BacktraceLogger.e(LOG_TAG, "Null action received. This is a bug");
+            return;
+        }
+
+        Map<String, Object> attributes = null;
+        Bundle extras = intent.getExtras();
+
+        if (extras != null) {
+            Set<String> keys = extras.keySet();
```

Comment:
Is this a reason to create keys if keys are null ? My proposal: 

```
if (extras != null && keys != null) {
            Set<String> keys = extras.keySet();
             attributes = new HashMap<String, Object>();
             ....
}
```

Reply:
Nice! Will do!